sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/planner/BeamRuleSets.java

```diff
@@ -82,8 +80,6 @@
           // Rules for window functions
           ProjectToWindowRule.PROJECT,
           // Rules so we only have to implement Calc
-          FilterCalcMergeRule.INSTANCE,
-          ProjectCalcMergeRule.INSTANCE,
```

Comment:
https://github.com/apache/beam/pull/12400/files I have done this part in ZetaSQL planner so you don't need to disable such rules in the default rulesets

Reply:
Meanwhile, one rule is required for ZetaSQL planner is, we probably need a `BeamZetaSQLCalcMergeRule`, which does not merge Calc when one Calc has Java UDF but another Calc does not. This will be useful when users write query like 
```
select java_udf(x) from (select builtin_function(y) as x from table) as t;
```

Reply:
I reverted changes to this file.